server/dive_server/utils.py

```diff
@@ -146,3 +156,62 @@ def saveCSVImportAttributes(folder, attributes, user):
 
     folder['meta']['attributes'] = attributes_dict
     Folder().save(folder)
+
+
+def verify_dataset(folder: GirderModel):
+    """Verify that a given folder is a DIVE dataset"""
+
+    if not asbool(fromMeta(folder, DatasetMarker, False)):
+        raise GirderException(f'Source folder is not a valid DIVE dataset')
+    return True
+
+
+def getCloneRoot(owner: GirderModel, source_folder: GirderModel):
+    """Get the source media folder associated with a clone"""
+    verify_dataset(source_folder)
+    next_id = fromMeta(source_folder, ForeignMediaIdMarker, False)
+    while next_id is not False:
+        """Recurse through source folders to find the root, allowing clones of clones"""
+        source_folder = Folder().load(
+            next_id,
+            level=AccessType.READ,
+            user=owner,
+        )
+        if source_folder is None:
+            raise GirderException(
+                f"Referenced media source missing. Folder Id {next_id} was not found."
+                " This may be a cloned dataset where the source was deleted."
+            )
+        verify_dataset(source_folder)
+        next_id = fromMeta(source_folder, ForeignMediaIdMarker, False)
+    return source_folder
+
+
+def createSoftClone(
+    owner: GirderModel,
+    source_folder: GirderModel,
+    parent_folder: GirderModel,
+    name: str = None,
+):
+    """Create a no-copy clone of folder with source_id for owner"""
+
+    cloned_folder = Folder().createFolder(
+        parent_folder,
+        name or source_folder['name'],
+        description=f'Clone of {source_folder["name"]}.',
+        reuseExisting=False,
+    )
+    cloned_folder['meta'] = source_folder['meta']
+    media_source_folder = getCloneRoot(owner, source_folder)
+    cloned_folder['meta'][ForeignMediaIdMarker] = str(media_source_folder['_id'])
```

Comment:
Clones of clones will always point directly to the root media, so clones don't have to be recursively resolved.